ThunderRequest/UIAlertController+ErrorRecovery.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  UIAlertController+ErrorRecovery.swift
+//  ThunderRequest
+//
+//  Created by Simon Mitchell on 14/12/2018.
+//  Copyright © 2018 threesidedcube. All rights reserved.
+//
+
+import UIKit
+
+extension ErrorRecoveryOption.Style {
+    var alertActionStyle: UIAlertAction.Style {
+        switch self {
+        case .cancel:
+            return .cancel
+        default:
+            return .default
+        }
+    }
+}
+
+public extension ErrorOverrides {
+    
+    /// Returns a summarised message body to display to the user combining
+    /// failure reasons and suggested recovery options if supplied
+    ///
+    /// - Parameter error: The error to format for
+    /// - Returns: An optional message
+    static func recoveryMessageFor(error: Error) -> String? {
+        
+        let recoverableError = error as? CustomisableRecoverableError
+        let override = ErrorOverrides.overrideFor(domain: recoverableError?.domain ?? (error as NSError).domain, code: recoverableError?.code ?? (error as NSError).code)
```

Comment:
Oooooh, didn't realise the term `override` could be used! Thought it was reserved but seems it's actually fine! :D

Reply:
Seems to be! Looks like it's just on GitHub that it's been picked up by syntax highlighting :P